src/Symfony/Component/Encryption/Phpseclib/PhpseclibEncryption.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption\Phpseclib;
+
+use phpseclib\Crypt\AES;
+use phpseclib\Crypt\Random;
+use phpseclib\Crypt\RSA;
+use Symfony\Component\Encryption\Ciphertext;
+use Symfony\Component\Encryption\EncryptionInterface;
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\EncryptionException;
+use Symfony\Component\Encryption\Exception\InvalidKeyException;
+use Symfony\Component\Encryption\Exception\UnableToVerifySignatureException;
+use Symfony\Component\Encryption\Exception\UnsupportedAlgorithmException;
+use Symfony\Component\Encryption\KeyInterface;
+
+if (!class_exists(RSA::class)) {
+    throw new \LogicException('You cannot use "Symfony\Component\Security\Core\Encryption\PhpseclibEncryption" as the "phpseclib/phpseclib:2.x" package is not installed. Try running "composer require phpseclib/phpseclib:^2".');
```

Comment:
this should be moved to the constructor instead. Otherwise, a `class_exists(PhpseclibEncryption::class)` could trigger it.

Reply:
Thank you